Framework/Nexus/src/MuonNexusReader.cpp

```diff
@@ -122,7 +122,48 @@ void MuonNexusReader::readFromFile(const string &filename) {
   // get instrument name
   handle.openGroup("instrument", "NXinstrument");
   handle.readData("name", nexus_instrument_name);
-  handle.closeGroup();
+
+  // Try to read in period information
+  std::vector<std::string> periodInformation{
```

Comment:
Perhaps this should go in a separate method to avoid this method getting too long?

For fixed length arrays I would also suggest using `std::array`, the C++ equivalent of fixed-length C arrays.
E.g.
```
const std::array<std::string, 7> periodInformation{"period_sequences",  "period_labels",  "period_type",  "frames_period_requested", "frames_period_raw", "total_counts_period", "period_output"};
```
If this is constant too then can we extract it as a "named module constant" in an anonymous namespace at the top.